static/js/src/certified-search-results.js

```diff
@@ -1,90 +1,389 @@
 
-  let objUrl = new URL(window.location);
-  const { href } = window.location;
-  if (href.includes("q=") && !href.includes("q=&")) {
-    const startOfQuery = href.indexOf("q");
-    const endOfQuery = href.indexOf("&");
-    const searchQuery = href.substring(startOfQuery, endOfQuery);
-    objUrl.search = searchQuery;
+// Hide more/less links when tabs are collapsed
+tabBtn2.addEventListener("click", (e) => {
+  if (tabBtn2.ariaExpanded === "true") {
+    showAllVendors.classList.add("u-hide");
+    showLessVendors.classList.add("u-hide");
+  }
+
+  if (tabBtn3.ariaExpanded === "true") {
+    showAllReleases.classList.add("u-hide");
+    showLessReleases.classList.add("u-hide");
+  }
+});
+
+// Set global filter limit for vendors and releases
+let filterLimit = 5;
+
+function loadFilters() {
+  const { category, vendor, release } = retrieveSelectedFilters();
+  renderFilters(category, vendor, release);
+}
+
+/**
+ *
+ * @returns {object} current state of filters in a flat object
+ *
+ * This function is used as state management
+ * It provides the current state of all filters (category, vendor and ralease)
+ */
+function retrieveSelectedFilters() {
+  const url = new URL(window.location.href);
+  const urlParams = new URLSearchParams(url.search);
+  const path = window.location.pathname;
+  const pathCategory = path.replace("/certified/", "");
+
+  let selectedCategories;
+  let selectedVendors = urlParams.getAll("vendor");
+  let selectedReleases = urlParams.getAll("release");
+
+  if (pathCategory !== "/certified") {
+    selectedCategories = urlParams.getAll("category").push(pathCategory);
   } else {
-    objUrl.search = "";
+    selectedCategories = urlParams.getAll("category");
   }
-  window.location.assign(objUrl);
-  return false;
+
+  return {
+    category: selectedCategories,
+    vendor: selectedVendors,
+    release: selectedReleases,
+  };
 }
 
-function enableApplyFilters() {
-  const filtersSelected = [];
-  const filters = document.querySelectorAll(".js-enable-apply-filters");
-  filters.forEach((filter) => {
-    if (filter.checked) {
-      filtersSelected.push(filter);
+function toggleFilterExpandLinks(data, total, elementMore, elementLess) {
+  if (data.length < total) {
+    // Not surpassing length so hide all
+    elementMore.classList.remove("u-hide");
+    elementLess.classList.add("u-hide");
+    return;
+  } else {
+    elementMore.classList.add("u-hide");
+    elementLess.classList.remove("u-hide");
+  }
+
+  if (data.length <= filterLimit) {
+    elementMore.classList.add("u-hide");
+    elementLess.classList.add("u-hide");
+  }
+}
+
+async function renderFilters(
+  categories,
+  vendors,
+  releases,
+  vendorLimit,
+  releaseLimit,
+  renderVendorFilters = true,
+  renderReleaseFilters = true
+) {
+  const filters = await fetchFilters(
+    categories,
+    vendors,
+    releases,
+    vendorLimit,
+    releaseLimit
+  );
+  if (categories && categories.length > 0) {
```

Comment:
looks like you don't need this condition since the code inside this `if` and its `else` is identical